### DISCUSSION TITLE
When running `dev`, `process.env.SSR` throw error?

### DISCUSSION THREAD

**Discussion by mapinxue** [solidjs-community/solid-primitives#249] [Q&A] [answered]
When I run `pnpm run start` in `solid-primitives/packages/selection`, it throw an error about process.env.SSR. I solved this error with `try..catch`. But is this normal?

## error
```js
index.ts:40 Uncaught ReferenceError: process is not defined
    at createSelection (index.ts:40:3)
    at _Hot$$App (index.tsx:7:37)
    at @solid-refresh:10:42
    at untrack (dev.js:421:12)
    at Object.fn (@solid-refresh:10:28)
    at runComputation (dev.js:693:22)
    at updateComputation (dev.js:676:3)
    at createMemo (dev.js:238:10)
    at hmrCompWrapper (@solid-refresh:7:20)
    at dev.js:515:12
```
## cause

```js
  if (process.env.SSR) {
    return [
      () => [null, NaN, NaN],
      sel => (typeof sel === "function" ? sel([null, NaN, NaN]) : sel)
    ];
  }
```

**Comment by thetarnav** [comment, marked as answer]
Oh, thank you for reporting this.
This is a bug, the build output shouldn't contain those process.env checks.
Will push a patch in a few hours

**Reply by thetarnav** [reply]
Ah, wait, this is not about the output, but local development, right?
In this case I guess I still have to update the vite config to define those env vars.
The vitest config is good, just need to copy what's there.

**Reply by mapinxue** [reply]
Yes, it's all about local development.

**Reply by thetarnav** [reply]
alright, fix should be on the main branch
